**Using Support Vector Machines on Artificial Characters dataset by Aana Kakroo (20BAI1138)**

In [1]:
# importing required Python libraries  
import numpy as np  
import matplotlib.pyplot as plt  
import pandas as pd  

Importing dataset into Jupyter notebook

In [2]:
#importing datasets  
df= pd.read_csv(r"C:\Users\aanak\OneDrive\Desktop\Materials for college\ML\LAB\artificial_char.csv") 
df 

V1  V2  V3  V4  V5     V6     V7  Class
0       0   0   0   0  20  20.00  46.10      1
1       1  19   0  19   8   8.00  46.10      1
2       2   0  20  19   8  22.47  46.10      1
3       3   0  20   8  42  23.41  46.10      1
4       4  19   8   8  42  35.74  46.10      1
...    ..  ..  ..  ..  ..    ...    ...    ...
10213   3  18  39  22  33   7.21  49.19      3
10214   4  22  33  22  20  13.00  49.19      3
10215   5  22  20  18  14   7.21  49.19      3
10216   6  18  14   0   8  18.97  49.19      3
10217   7   0   8  20   0  21.54  49.19      3

[10218 rows x 8 columns]

Checking datatypes of values in each column in the dataframe

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10218 entries, 0 to 10217
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   V1      10218 non-null  int64  
 1   V2      10218 non-null  int64  
 2   V3      10218 non-null  int64  
 3   V4      10218 non-null  int64  
 4   V5      10218 non-null  int64  
 5   V6      10218 non-null  float64
 6   V7      10218 non-null  float64
 7   Class   10218 non-null  int64  
dtypes: float64(2), int64(6)
memory usage: 638.8 KB


As other classifiers, SVC takes as input two arrays: an array X of shape (n_samples, n_features) holding the training samples, and an array y of class labels (strings or integers), of shape (n_samples)

In [4]:
x = df.drop('Class',axis=1)
y = df['Class']

Change x,y from dataframes to arrays so that they can be used as parameters in SVC

In [5]:
x=np.array(x)
y=np.array(y)

Building SVC model with default parameters and gamma=auto

In [6]:
#importing required libraries
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC

#building the model
clf = make_pipeline(StandardScaler(), SVC(gamma='auto'))
clf.fit(x, y)
clf.score(x,y)

0.6476805637110981

Building SVC model with C=2.0

In [7]:
clf2 = make_pipeline(StandardScaler(), SVC(C=2.0))
clf2.fit(x, y)
clf2.score(x,y)

0.6629477392836172

Building SVC model with C=3.0

In [8]:
clf3 = make_pipeline(StandardScaler(), SVC(C=3.0))
clf3.fit(x, y)
clf3.score(x,y)

0.6758661186142102

Building SVC model with kernel='linear'

In [9]:
clf4 = make_pipeline(StandardScaler(), SVC(kernel='linear'))
clf4.fit(x, y)
clf4.score(x,y)

0.44901154824818945

Building SVC model with kernel='poly'

In [10]:
clf5 = make_pipeline(StandardScaler(), SVC(kernel='poly'))
clf5.fit(x, y)
clf5.score(x,y)

0.6022705030338618

Building SVC model with kernel='sigmoid'

In [11]:
clf5 = make_pipeline(StandardScaler(), SVC(kernel='sigmoid'))
clf5.fit(x, y)
clf5.score(x,y)

0.18301037385006852

Building SVC model with default parameters

In [12]:
clf7 = make_pipeline(StandardScaler(), SVC())
clf7.fit(x, y)
clf7.score(x,y)

0.6476805637110981

Building SVC model with kernel='poly' and degree=4

In [13]:
clf8 = make_pipeline(StandardScaler(), SVC(kernel='poly',degree=4))
clf8.fit(x, y)
clf8.score(x,y)

0.6187120767273439

Building SVC model with C=5.0

In [14]:
clf9 = make_pipeline(StandardScaler(), SVC(C=5.0))
clf9.fit(x, y)
clf9.score(x,y)

0.6864357017028773

Building SVC model with C=5.0

In [15]:
clf10 = make_pipeline(StandardScaler(), SVC(C=10.0))
clf10.fit(x, y)
clf10.score(x,y)

0.6979839498923468

The highest accuracy score attained so far from SVC models is 0.6979839498923468 which was obtained from using C=5.0 as the parameter. As observed, by increasing the value of C but keeping rest of the parameters as default, the accuracy score of the model increased.